src/server/editorServices.ts

```diff
@@ -717,59 +740,38 @@ namespace ts.server {
-                    this.openFiles.push(info);
-                }
-                return;
-            }
-
-            let foundConfiguredProject = false;
-            for (const p of info.containingProjects) {
-                // file is the part of configured project
-                if (p.projectKind === ProjectKind.Configured) {
-                    foundConfiguredProject = true;
-                    if (addToListOfOpenFiles) {
-                        ((<ConfiguredProject>p)).addOpenRef();
-                    }
-                }
-            }
-            if (foundConfiguredProject) {
-                if (addToListOfOpenFiles) {
-                    this.openFiles.push(info);
-                }
-                return;
-            }
-
             if (info.containingProjects.length === 0) {
                 // create new inferred project p with the newly opened file as root
                 // or add root to existing inferred project if 'useOneInferredProject' is true
-                const inferredProject = this.createInferredProjectWithRootFileIfNecessary(info);
-                if (!this.useSingleInferredProject) {
-                    // if useOneInferredProject is not set then try to fixup ownership of open files
-                    // check 'defaultProject !== inferredProject' is necessary to handle cases
-                    // when creation inferred project for some file has added other open files into this project (i.e. as referenced files)
-                    // we definitely don't want to delete the project that was just created
-                    for (const f of this.openFiles) {
-                        if (f.containingProjects.length === 0 || !inferredProject.containsScriptInfo(f)) {
-                            // this is orphaned file that we have not processed yet - skip it
-                            continue;
-                        }
-
-                        for (const fContainingProject of f.containingProjects) {
-                            if (fContainingProject.projectKind === ProjectKind.Inferred &&
-                                fContainingProject.isRoot(f) &&
-                                fContainingProject !== inferredProject) {
-
-                                // open file used to be root in inferred project,
-                                // this inferred project is different from the one we've just created for current file
-                                // and new inferred project references this open file.
-                                // We should delete old inferred project and attach open file to the new one
-                                this.removeProject(fContainingProject);
-                                f.attachToProject(inferredProject);
-                            }
+                this.createInferredProjectWithRootFileIfNecessary(info);
+
+                // if useOneInferredProject is not set then try to fixup ownership of open files

```

Comment:
Typo: "useSingleInferredProject", not "useOneInferredProject"
Also, the formatting here makes it hard to tell what's the start of a sentence and what's a continuation. Could you add some punctuation and capitalization?
Also, if `createInferredProjectWithRootFileIfNecessary` is getting us into a bad state, it shouldn't be the caller's responsibility to fixup everything; this loop should be in `createInferredProjectWithRootFileIfNecessary`. But I don't see why adding a new inferred project should require us to loop over every other inferred project and check for errors.